crates/sui-types/src/sui_system_state.rs

```diff
@@ -277,6 +283,63 @@ impl SuiSystemState {
         }
     }
 
+    pub fn get_current_epoch_peer_ids(&self) -> HashMap<PeerId, String> {
+        let mut result = HashMap::new();
+        let primary_peer_ids: Vec<PeerId> = self
+            .validators
+            .active_validators
+            .iter()
+            .map(|validator| {
+                let network_key = narwhal_crypto::NetworkPublicKey::from_bytes(
+                    &validator.metadata.network_pubkey_bytes,
+                )
+                .expect("Can't get narwhal network key");
```

Comment:
Probably we should have a message closer to SUI? ex `Can't get network key`

Reply:
More reasonable message is probably something like `"Malformed network public key"`.